packages/flutter/lib/src/services/text_input.dart

```diff
@@ -896,17 +923,24 @@ class TextInput {
   /// A client that no longer wishes to interact with the text input control
   /// should call [TextInputConnection.close] on the returned
   /// [TextInputConnection].
-  static TextInputConnection attach(TextInputClient client, TextInputConfiguration configuration) {
+  static TextInputConnection attach(TextInputClient client) {
     assert(client != null);
-    assert(configuration != null);
-    assert(_debugEnsureInputActionWorksOnPlatform(configuration.inputAction));
     final TextInputConnection connection = TextInputConnection._(client);
     _clientHandler._currentConnection = connection;
-    SystemChannels.textInput.invokeMethod<void>(
+    _attach(_clientHandler, connection);
+    return connection;
+  }
+
+  static void _attach(TextInputClientHandler handler, TextInputConnection connection) {
```

Comment:
There should probably be some documentation on what the difference between attach and _attach is.

Reply:
Done